packages/schemas/alterations/next-1670991222-terms-of-use.ts

```diff
@@ -0,0 +1,86 @@
+import type { DatabaseTransactionConnection } from 'slonik';
+import { sql } from 'slonik';
+
+import type { AlterationScript } from '../lib/types/alteration.js';
+
+type DeprecatedTermsOfUse = {
+  enabled: boolean;
+  contentUrl?: string;
+};
+
+type DeprecatedSignInExperience = {
+  id: string;
+  termsOfUse: DeprecatedTermsOfUse;
+};
+
+type SignInExperience = {
+  id: string;
+  termsOfUseUrl?: string | null;
+};
+
+const alterTermsOfUse = async (
+  signInExperience: DeprecatedSignInExperience,
+  pool: DatabaseTransactionConnection
+) => {
+  const {
+    id,
+    termsOfUse: { enabled, contentUrl },
+  } = signInExperience;
+
+  if (enabled && contentUrl) {
+    await pool.query(
+      sql`update sign_in_experiences set terms_of_use_url = ${contentUrl} where id = ${id}`
+    );
+  }
+};
+
+const rollbackTermsOfUse = async (
+  signInExperience: SignInExperience,
+  pool: DatabaseTransactionConnection
+) => {
+  const { id, termsOfUseUrl } = signInExperience;
+
+  const termsOfUse: DeprecatedTermsOfUse = {
+    enabled: Boolean(termsOfUseUrl),
+    contentUrl: termsOfUseUrl ?? '',
+  };
+
+  await pool.query(
+    sql`update sign_in_experiences set terms_of_use = ${JSON.stringify(
+      termsOfUse
+    )} where id = ${id}`
+  );
+};
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    const rows = await pool.many<DeprecatedSignInExperience>(
+      sql`select * from sign_in_experiences`
+    );
+
+    await pool.query(sql`
+      alter table sign_in_experiences add column terms_of_use_url varchar(2048)
```

Comment:
any reference or standard are we following about the URL varchar size? Will it break if the old data contains URL length that is longer than 2048?

Reply:
Yes, URLs in our DB are all in 2048 size.
The old URL is stored in JSONB and has no size limitation, but I think 2048 is long enough for an URL.

Reply:
<img width="645" alt="image" src="https://user-images.githubusercontent.com/10806653/207775026-c3931bf5-2d8d-4770-8499-599dc22d764d.png">

Reply:
By the way, If the original URL is longer than 2048, this alteration will fail; this is what we expect.